streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -431,7 +450,8 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
           receiver.preferredLocation,
           receiverTrackingInfos,
           getExecutors)
-        updateReceiverScheduledExecutors(receiver.streamId, scheduledExecutors)
```

Comment:
@zsxwing explained me this offline. In case of resecheduling, the scheduled executors are not stored so that the following scenario does not occur.
- Initial globally-optimal schedule is stored, but one receiver gets launched incorrectly. 
- The receiver is rejected and therefore has to be rescheduled, but if the rescheduled location (which is locally-optimal for that receiver) is saved, it will overwrite the original globally optimal location, and will get launched somewhere else that does not ensure the proper global balancing.